### PR TITLE
add filter argument to assertDumpEquals()

### LINKED ISSUE #8172: [VarDumper] Add filter in VarDumperTestTrait
See https://github.com/symfony/symfony/pull/22588.

`assertDumpEquals()` and `assertDumpMatchesFormat()` now define a new argument to filter things. Check if we use them in any code sample.

### PR DESCRIPTION
**Pull request by xabbuh**
This fixes #8172.

### CODE REVIEW

File: components/var_dumper/advanced.rst [hunk 1]
```diff
@@ -206,7 +206,7 @@ The output format of a dumper can be fine tuned by the two flags
 ``DUMP_STRING_LENGTH`` and ``DUMP_LIGHT_ARRAY`` which are passed as a bitmap
 in the third constructor argument. They can also be set via environment 
 variables when using
-:method:`assertDumpEquals($dump, $data, $message) <Symfony\\Component\\VarDumper\\Test\\VarDumperTestTrait::assertDumpEquals>`
+:method:`assertDumpEquals($dump, $data, $filter, $message) <Symfony\\Component\\VarDumper\\Test\\VarDumperTestTrait::assertDumpEquals>`
```
Comment (HeahDude) [inline components/var_dumper/advanced.rst]: No version added directive nor explanation of the purpose of this argument?

**Comment by javiereguiluz** [comment]
@xabbuh do you think we need to add more information about this change as Jules asked for or is it enough as it is now? Thanks!

**Comment by xabbuh** [comment]
I added some more information.

**Review by javiereguiluz — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you Christian.

**Pull request merged**